### PR TITLE
Fix events on rocks

### PR DESCRIPTION
**Pull request by mafintosh**

### CODE REVIEW

File: lib/core.js [hunk 1]
```diff
@@ -130,26 +120,27 @@ class Update {
...
     this.name = name
+    this.sessions = []
 
     this.mutex = new Mutex()
 
     this.blocks = blocks
     this.tree = tree
     this.bitfield = bitfield
     this.snapshotLength = snapshotLength
+    this.snapshotCompatLength = snapshotLength
```
Comment (chm-diederichs) [inline lib/core.js:143]: do we still need snapshotCompatLength separately?

File: lib/core.js [hunk 2]
```diff
@@ -1302,71 +1352,13 @@ module.exports = class Core {
...
-    }
-  }
-
   async _onconflict (proof, from) {
     await this.replicator.onconflict(from)
 
-    for (const s of this.sessions) s.emit('conflict', proof.upgrade.length, proof.fork, proof)
+    for (let i = this.monitors.length - 1; i >= 0; i--) {
+      const s = this.monitors[i]
+      s.emit('conflict', proof.upgrade.length, proof.fork, proof)
```
Comment (chm-diederichs) [inline lib/core.js:1360]: shouldn't this be emitted by all sessions?

File: lib/core.js (hunk 1 above)
Comment (mafintosh) [reply lib/core.js:143]: ya cause the normal one is used for the actual length and the compat is how much you can reuse (i think)

File: lib/core.js (hunk 2 above)
Comment (chm-diederichs) [reply lib/core.js:1360]: ahh sorry i misread the first line of maybeAddMonitor

**Review by chm-diederichs — APPROVED** [review]

**Pull request merged**